src/handlers/comment/handlers/review.ts

```diff
@@ -0,0 +1,196 @@
+ */
+export const review = async (body: string) => {
+  const context = getBotContext();
+  const logger = getLogger();
+
+  const payload = context.payload as Payload;
+  const issue = payload.issue;
+
+  if (!issue) {
+    return `Payload issue is undefined`;
+  }
+
+  if (!body) {
+    return `Payload body is undefined`;
+  }
+
+  const isPr = await getPullByNumber(context, issue.number);
+
+  if (!isPr) {
+    return `This command can only be used on pull requests.`;
+  }
+
+  const reviewRegex = /^\/review/;
+  const reviewRegexMatch = body.match(reviewRegex);
+
+  if (!reviewRegexMatch) {
+    return `Payload body does not match regex`;
+  }
+
+  const streamlined: StreamlinedComment[] = [];
+  let chatHistory: CreateChatCompletionRequestMessage[] = [];
+  let linkedPRStreamlined: StreamlinedComment[] = [];
+  let linkedIssueStreamlined: StreamlinedComment[] = [];
+
+  const comments = await getAllIssueComments(issue.number);
+  const commentsRaw = await getAllIssueComments(issue.number, "raw");
+
+  if (!commentsRaw) {
+    logger.info(`Error getting issue comments`);
+    return `Error getting issue comments`;
+  }
+  streamlined.push({
+    login: issue.user.login,
+    body: issue.body,
+  });
+
+  comments.forEach(async (comment, i) => {
+    if (comment.user.type == UserType.User || commentsRaw[i].body.includes("<!--- { 'UbiquityAI': 'answer' } --->")) {
+      streamlined.push({
+        login: comment.user.login,
+        body: comment.body,
+      });
+    }
+  });
+
+  // returns the conversational context from all linked issues and prs
+  const links = await getAllLinkedIssuesAndPullsInBody(issue.number);
+
+  if (typeof links === "string") {
+    logger.info(`Error getting linked issues or prs: ${links}`);
+    return `Error getting linked issues or prs: ${links}`;
+  }
+  linkedIssueStreamlined = links.linkedIssues;
+  linkedPRStreamlined = links.linkedPrs;
+
+  // return a diff of the changes made in the PR
+  const comparePR = async () => {
+    const comparePR = await context.octokit.pulls.get({
+      owner: payload.repository.owner.login,
+      repo: payload.repository.name,
+      pull_number: issue.number,
+    });
+
+    const pr = comparePR.data;
+
+    const prDiff = await context.octokit.pulls.get({
+      owner: payload.repository.owner.login,
+      repo: payload.repository.name,
+      pull_number: pr.number,
+      mediaType: {
+        format: "diff",
+      },
+    });
+
+    const diffContent = prDiff.data;
+
+    return {
+      pr,
+      diff: diffContent,
+    };
+  };
+
+  const isPull = async () => {
+    if (isPr) {
+      const diff = await comparePR()
+        .then(({ diff }) => {
+          return diff;
+        })
+        .catch((error) => {
+          logger.info(`Error getting diff: ${error}`);
+          return `Error getting diff: ${error}`;
+        });
+
+      chatHistory.push({
+        role: "system",
+        content: gptContextTemplate,
+        name: "UbiquityAI",
```

Comment:
What is the name field for

Reply:
I'll remove them since we are not using functions.

```ts
export interface ChatCompletionMessageParam {
  /**
   * The contents of the message. `content` is required for all messages, and may be
   * null for assistant messages with function calls.
   */
  content: string | null;

  /**
   * The role of the messages author. One of `system`, `user`, `assistant`, or
   * `function`.
   */
  role: 'system' | 'user' | 'assistant' | 'function';

  /**
   * The name and arguments of a function that should be called, as generated by the
   * model.
   */
  function_call?: ChatCompletionMessageParam.FunctionCall;

  /**
   * The name of the author of this message. `name` is required if role is
   * `function`, and it should be the name of the function whose response is in the
   * `content`. May contain a-z, A-Z, 0-9, and underscores, with a maximum length of
   * 64 characters.
   */
  name?: string;
}
```